### PR TITLE
Issue 536 - Migrate REST API to v4

### PR DESCRIPTION
**Pull request by wardcr**
- Changed API to require a version prefix. This uncovered an incompatibility between URL prefixes and namespaces that isn't really documented. A workaround was developed by grouping the versioned APIs into separate URL pattern groups when the master URL routes are registered. This also required updating every REST unit test to include the latest version prefix by default. It also required using a DRF method for reversing URLs instead of the standard Django one.

- Updated the logs API to support the v3/v4 behavior.

- Updated the job details and recipe details to support v3/v4 behavior.

- Updated docs for v3/v4 changes.

- Added a TODO tag "API_V3" throughout the code to make future clean up work easier.

### CODE REVIEW

File: scale/docs/rest/job.rst [hunk 1]
```diff
@@ -198,6 +198,8 @@ These services provide access to information about "all", "currently running" an
 
 .. _rest_job_details:
 
+.. TODO: API_V3 Remove attributes
```
Comment (dfaller) [inline scale/docs/rest/job.rst]: Still work to do here?

File: scale/docs/rest/job_execution.rst [hunk 2]
```diff
@@ -353,6 +353,7 @@ These services provide access to information about "all", "currently running" an
 |  }                                                                                                                        |
 +---------------------------------------------------------------------------------------------------------------------------+
 
+.. TODO: API_V3 Remove this block
```
Comment (dfaller) [inline scale/docs/rest/job_execution.rst]: Remove this line? It mentions being gone in v4.

File: scale/docs/rest/recipe.rst [hunk 3]
```diff
@@ -137,6 +137,8 @@ These services provide access to information about recipes.
 
 .. _rest_recipe_details:
 
+.. TODO: API_V3 Remove attributes
```
Comment (dfaller) [inline scale/docs/rest/recipe.rst]: Do we want to document the old v3 fields?

**Comment by wardcr** [comment]
Closes #556

**Pull request merged**